src/pages/api/searchOperators.ts

```diff
@@ -2,17 +2,71 @@ import { NextApiResponse } from 'next';
 import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
 import { SEARCH_OPERATOR_ATTRIBUTE } from '../../constants';
 import { redirectTo, redirectToError } from './apiUtils';
-import { updateSessionAttribute } from '../../utils/sessions';
+import { getSessionAttribute, updateSessionAttribute } from '../../utils/sessions';
 import { removeExcessWhiteSpace } from './apiUtils/validator';
+import { OperatorNameType } from '../../data/auroradb';
 
-// unsure if this interface is needed
-export interface SearchOperatorsWithErrors {
+export interface SearchOperators {
+    selectedOperators: OperatorNameType[];
     errors: ErrorInfo[];
 }
 
 export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
     const errors: ErrorInfo[] = [];
 
+    const { addOperator } = req.body;
+    const refererUrl = req?.headers?.referer;
+    const queryString = refererUrl?.substring(refererUrl?.indexOf('?') + 1);
+
+    const searchOperatorAttribute = getSessionAttribute(req, SEARCH_OPERATOR_ATTRIBUTE);
+
+    if (addOperator) {
+        const requestBody: { [key: string]: string | string[] } = req.body;
+
+        if (Object.keys(req.body).length - 2 === 0) {
+            errors.push({
+                errorMessage: 'Choose at least one operator from the options',
+                id: 'searchText',
+            });
+
+            updateSessionAttribute(req, SEARCH_OPERATOR_ATTRIBUTE, {
+                errors,
+                selectedOperators: searchOperatorAttribute?.selectedOperators || [],
+            });
+            redirectTo(res, `/searchOperators?${queryString}`);
+        }
+
+        const selectedOperators: OperatorNameType[] = [];
+
+        const currentSelectedOperators = (searchOperatorAttribute && searchOperatorAttribute.selectedOperators) || [];
+
+        Object.entries(requestBody).forEach(entry => {
```

Comment:
Might be worth moving this out to a method, as I cant really understand what it's doing, and it having a name might help someone understand - also if we move it out to a function, we can test it.